### PR TITLE
Task #388. Implement Direct Messaging on Client side.

### LINKED ISSUE #388: Implement Direct Messaging on Client side.
Depends on #385 

As a user I want to message other users directly.

Client side should support several things for this feature to work:
* Clicking on the username in the existing channel opens a direct chat with that user.
* DM channels looks and behave in the same way as channels do.
* DM channels have their own history and according server requests must be done.
* DM channel is named the same as the username

AC: Clicking on any username opens up the DM channel with that user.

### PR DESCRIPTION
**Pull request by ivanboika**
Task: 
Depends on https://github.com/f1nal3/Juniorgram/issues/385

As a user I want to message other users directly.

Client side should support several things for this feature to work:

Clicking on the username in the existing channel opens a direct chat with that user.
DM channels looks and behave in the same way as channels do.
DM channels have their own history and according server requests must be done.
DM channel is named the same as the username
AC: Clicking on any username opens up the DM channel with that user.

My addition to task:
image

Also, I will try to change style of the button if it's necessary. 
P.S Besides that, This task requires changes in DB system.

### CODE REVIEW

File: Client/Client.Qt/Widgets/MessageWidget.cpp [hunk 1]
```diff
@@ -67,7 +75,8 @@ void MessageWidget::paintEvent(QPaintEvent* e)
     }
 
     auto usernameRect = QRect(margin * 2, margin * 2 + 1, getStyle().fontname->width(getUserName()), getStyle().fontname->height);
-    p.drawText(usernameRect, getUserName());
+
+    // p.drawText(usernameRect, getUserName());
```
Comment (memorisecodead) [inline Client/Client.Qt/Widgets/MessageWidget.cpp:79]: I would get rid of the commented lines that are not used in the code

File: Client/Client.Qt/Widgets/ChatWidget.cpp [hunk 2]
```diff
@@ -82,7 +82,7 @@ void ChatWidget::setReply(QString messageText, QString username, uint64_t messag
     _replyWidget->show();
 }
 
-void ChatWidget::updateLayout()
+void ChatWidget::updateLayout() // at that point we repaint everything
```
Comment (memorisecodead) [inline Client/Client.Qt/Widgets/ChatWidget.cpp:85]: Repainting? Not quite sure what you mean, but I'm guessing

File: DataAccess/DataAccess.Postgre/PostgreRepositories.cpp [hunk 3]
```diff
@@ -473,30 +506,6 @@ std::optional<pqxx::result> RepliesRepository::insertIDsIntoChannelRepliesTable(
...
-        auto cId       = std::any_cast<pqxx::result>(result.value());
-        auto channelId = std::to_string(cId[0][0].as<uint64_t>());
-        adapter->query("INSERT INTO user_channels VALUES (" + minUserId + "," + channelId + "), (" + maxUserId + "," + channelId +
-                       ")ON CONFLICT DO NOTHING;");
-        return Utility::DirectMessageStatus::SUCCESS;
-    }
-    return Utility::DirectMessageStatus::FAILED;
-}
 
 }  // namespace DataAccess
```
Comment (memorisecodead) [inline DataAccess/DataAccess.Postgre/PostgreRepositories.cpp:511]: Again, I think there should be no commented out parts of the code that are not used :)

**Review by memorisecodead — COMMENTED** [review]
Great, but tweak the subtleties and you'll be just a sweetie.

File: Client/Client.Qt/Widgets/ChatWidget.cpp (hunk 2 above)
Comment (ivanboika) [reply Client/Client.Qt/Widgets/ChatWidget.cpp:85]: Maybe it's mistake and this comment should be in the another place, but task requires to paint our new direct chat and I did it for myself just not to forget about it

File: Client/Client.Qt/Widgets/MessageWidget.cpp (hunk 1 above)
Comment (ivanboika) [reply Client/Client.Qt/Widgets/MessageWidget.cpp:79]: I know, but that's a draft request. There will no be commented lines like this when PR is ready for review

Comment (Orange-hanter) [reply Client/Client.Qt/Widgets/MessageWidget.cpp:79]: While this is a draft, it's okey

**Pull request closed without merging**